docs/sources/flow/reference/components/pyroscope.ebpf.md

```diff
@@ -0,0 +1,250 @@
+- `/lib/x86_64-linux-gnu/libc.so.6.debug`
+- `/lib/x86_64-linux-gnu/.debug/libc.so.6.debug`
+- `/usr/lib/debug/lib/x86_64-linux-gnu/libc.so.6.debug`
+
+### Dealing with Unknown Symbols
+
+If you encounter unknown symbols in the profiles you’ve collected, this indicates that we couldn't access an ELF file
+associated to a given address in trace. This might be due to the ELF file being inaccessible, or the absence of an entry
+in `/proc/pid/maps` for a particular address (this can be the case with JIT-compiled functions).
+
+This can occur for a number of reasons:
+
+- The process has terminated, making the ELF file inaccessible.
+- The ELF file is either corrupted or not recognized as an ELF file.
+- There is no corresponding ELF file entry in /proc/pid/maps for the address in the stack trace.
+
+### Addressing Unresolved Symbols
+
+If you only see module names (e.g., `/lib/x86_64-linux-gnu/libc.so.6`) without corresponding function names, this
+indicates that the symbols couldn't be mapped to their respective function names.
+
+This can occur for a number of reasons:
+
+- The binary has been stripped, leaving no .symtab, .dynsym, or .gopclntab sections in the ELF file.
+- The debug file is missing or could not be located.
+
+To fix this for your binaries, ensure that they either not stripped or that you have separate
+debug files available. You can achieve this by running:
+
+```bash
+objcopy --only-keep-debug elf elf.debug
+strip elf -o elf.stripped
+objcopy --add-gnu-debuglink=elf.debug elf.stripped elf.debuglink
+```
+
+For system libraries, ensure that debug symbols are installed. On Ubuntu, for example, you can install them by
+executing:
+
+```bash
+apt install libc6-dbg
+```
+
+### Understanding Flat Stack Traces
+
+If your profiles show a lof of shallow stack traces, typically 1-2 frames deep, it might mean that
+your binary was compiled without frame pointers.
+
+To compile your code with frame pointers, include the `-fno-omit-frame-pointer` flag in your compiler options.
+
+### Profiling Interpreted Languages
+
+Currently, profiling interpreted languages like Python, Ruby, JavaScript, etc., is not ideal using this implementation.
+The JIT-compiled methods in these languages are typically not in ELF file format, demanding additional steps for
+profiling. For instance, using perf-map-agent and enabling frame pointers for Java.
+
+Interpreted methods will display the interpreter function’s name, rather than the actual function. Future releases of
+the profiler are expected to support profiling of Ruby and Python code more effectively.
+
+## Example
+
+### Kubernetes discovery
+
+In the following example, performance profiles are collected from pods on the same node, discovered using
+`discovery.kubernetes`. Pod selection relies on the `HOSTNAME` environment variable, which is a pod name if the agent is
+used as a Grafana agent helm chart. Service name is set to `{namespace}/{container_name}` from kubernetes meta labels.
```

Comment:
```suggestion
used as a Grafana agent helm chart. The service name is set to `{namespace}/{container_name}` from Kubernetes meta labels.
```

Is Service name a variable? If it is a variable, we need to format it correctly here. Looking at the Docker example it's called "the `service_name` label"... is it the same thing here.

Is it line 204 that is being referred to here? If so there seems to be a disconnect between 

> Service name is set to `{namespace}/{container_name}`

and this line from the example:

> source_labels = ["__meta_kubernetes_namespace", "__meta_kubernetes_pod_container_name"]

Reply:
> Is Service name a variable?

Service name is a label -  `service_name`.

> Is it line 204 that is being referred to here?

Yes

> If so there seems to be a disconnect between

Do you mean it should be `{__meta_kubernetes_namespace}/{__meta_kubernetes_pod_container_name}`

Reply:
I've replaced the sentence with 

The `service_name` label is set to `{__meta_kubernetes_namespace}/{__meta_kubernetes_pod_container_name}` from kubernetes meta labels.